components/RepeatModal.tsx

```diff
@@ -0,0 +1,201 @@
+    endDate: Date,
+    count: number
+  ) => void;
+  // props for initial state values
+  initRepeat: boolean;
+  initInterval: number;
+  initWeekDays: number[];
+  initEndSelection: string;
+  initEndDate: Date;
+  initCount: number;
+};
+
+const RepeatModal: React.FC<RepeatModalProps> = ({
+  handleClose,
+  handleStates,
+  initRepeat,
+  initInterval,
+  initWeekDays,
+  initEndSelection,
+  initEndDate,
+  initCount,
+}) => {
+  // repeat modal states initialized by init props
+  const [repeat, setRepeat] = useState(initRepeat);
+  const [interval, setInterval] = useState(initInterval);
+  const [weekDays, setWeekDays] = useState<number[]>(initWeekDays);
+  const [endSelection, setEndSelection] = useState(initEndSelection);
+  const [endDate, setEndDate] = useState(initEndDate);
+  const [count, setCount] = useState(initCount);
+  const [valid, setValid] = useState(false);
+
+  // validate modal on field input
+  useEffect(() => {
+    setValid(true);
+    if (repeat) {
+      if (!interval) {
+        setValid(false);
+      }
+      if (endSelection === "on" && !endDate) {
+        setValid(false);
+      }
+      if (endSelection === "after" && !count) {
+        setValid(false);
+      }
+    }
+  }, [repeat, interval, weekDays, endSelection, endDate, count]);
+
+  // display strings for week days
+  const daysOfWeek = ["S", "M", "T", "W", "T", "F", "S"];
+
+  // adds/removes day from weekDay state
+  const handleDayOfWeekChange = (day: number): void => {
+    if (weekDays.includes(day)) {
+      setWeekDays(weekDays.filter((item) => item !== day));
+    } else {
+      setWeekDays((weekDays) => [...weekDays, day]);
+    }
+  };
+
+  // converts daysOfWeek idx to RRule weekday idx (0 == RRule.MO, ...)
+  const convertDayIdx = (idx: number): number => {
+    return (((idx - 1) % 7) + 7) % 7;
+  };
+
+  // passes state values to callback and closes modal on save
+  const handleSave = (): void => {
+    handleStates(repeat, interval, weekDays, endSelection, endDate, count);
+    handleClose();
+  };
+
+  return (
+    <div className={styles.modalWrapper}>
+      <div className={styles.modalContent}>
+        <div className={styles.headerWrapper}>
+          <p className={styles.title}>Repeat</p>
+          <div>
+            <label className={styles.switch}>
+              <input type="checkbox" checked={repeat} onChange={() => setRepeat(!repeat)} />
+              <span className={`${styles.slider} ${styles.round}`}></span>
+            </label>
+          </div>
+        </div>
+
+        <div className={styles.row}>
+          <p className={styles.largeLabel}>Repeat every</p>
+          <input
+            className={styles.numberInput}
+            disabled={!repeat}
+            type="number"
+            min={1}
+            value={interval}
+            onChange={(e) => setInterval(e.target.valueAsNumber)}
+          />
+          <input className={styles.freqField} disabled={!repeat} value="weekly" readOnly />
```

Comment:
These have to repeat every one week now :). So there is no need to have the input anymore. Just display something like "repeats weekly"